### DISCUSSION TITLE
Task 2

### DISCUSSION THREAD

**Discussion by VijayDamodharan** [Part-II-Computational-Physics/.github#42] [Q&A]
Hi, I have the following code for task 2:

`def CompleteHamiltonian2(N, h=1.):
    Hloc = LocalHamiltonian(h)
    H = np.zeros((2**N, 2**N))
    
    for i in range(N):
        H -= ft.reduce(np.kron, Hloc)
        Hloc -= np.roll(Hloc, 1, axis=0)
    
    return H
`

Since I basically thought that the final Hamiltonian is just a sum of the local hamiltonians. But I get the following error: 

`ValueError                                Traceback (most recent call last)
Cell In[41], line 12
      8         Hloc -= np.roll(Hloc, 1, axis=0)
     10     return H
---> 12 CompleteHamiltonian2(10, h = 1.23)

Cell In[41], line 7, in CompleteHamiltonian2(N, h)
      5 iterations = int(N/4)
      6 for i in range(iterations):
----> 7     H -= ft.reduce(np.kron, Hloc)
      8     Hloc -= np.roll(Hloc, 1, axis=0)
     10 return H

ValueError: operands could not be broadcast together with shapes (1024,1024) (256,) (1024,1024) `

Not sure why this is the case. Any help would be appreciated, thanks :)

**Comment by JNHallen** [comment]
Hi! Take a look at my answer to a similar questions [here](https://github.com/orgs/Part-II-Computational-Physics/discussions/40#discussion-4915300).
